### ISSUE TITLE
exporting an anonymous default function crashes tslint

### ISSUE THREAD

**Issue by PaulAvery** [palantir/tslint#456]
Exactly like the title says,

```ts
export default function() {}
```

results in the following crash:
```
var variableName = node.name.text;
                                    ^
TypeError: Cannot read property 'text' of undefined
    at NoUnusedVariablesWalker.visitFunctionDeclaration (/home/florian/dev/modules/node-kodi-remote/node_modules/tslint/build/rules/noUnusedVariableRule.js:118:37)
    at NoUnusedVariablesWalker.SyntaxWalker.visitNode (/home/florian/dev/modules/node-kodi-remote/node_modules/tslint/bin/tslint-cli.js:41519:26)
    at /home/florian/dev/modules/node-kodi-remote/node_modules/tslint/bin/tslint-cli.js:41639:67
    at visitEachNode (/home/florian/dev/modules/node-kodi-remote/node_modules/tslint/bin/tslint-cli.js:5928:30)
    at Object.forEachChild (/home/florian/dev/modules/node-kodi-remote/node_modules/tslint/bin/tslint-cli.js:6070:24)
    at NoUnusedVariablesWalker.SyntaxWalker.walkChildren (/home/florian/dev/modules/node-kodi-remote/node_modules/tslint/bin/tslint-cli.js:41639:16)
    at NoUnusedVariablesWalker.SyntaxWalker.visitSourceFile (/home/florian/dev/modules/node-kodi-remote/node_modules/tslint/bin/tslint-cli.js:41412:18)
    at NoUnusedVariablesWalker.SyntaxWalker.visitNode (/home/florian/dev/modules/node-kodi-remote/node_modules/tslint/bin/tslint-cli.js:41603:26)
    at NoUnusedVariablesWalker.SyntaxWalker.walk (/home/florian/dev/modules/node-kodi-remote/node_modules/tslint/bin/tslint-cli.js:41253:18)
    at Rule.AbstractRule.applyWithWalker (/home/florian/dev/modules/node-kodi-remote/node_modules/tslint/bin/tslint-cli.js:41768:24)
```

I think the fix should be pretty straightforward and I would submit a pull request myself, but in its current state I cannot get the codebase to build, so I cannot really go about fixing it. This was tested in `tslint@2.3.0-beta` straight from npm.

**Comment by leeavital** [comment]
What rules are you using?

**Comment by adidahiya** [comment]
looks like `no-unused-variable` rule is crashing

**Comment by ashwinr** [comment]
:(

**Comment by lukeapage** [comment]
any chance of a release? this is catching us out...

**Comment by adidahiya** [comment]
@lukeapage we're likely going to make a release today which includes a fix for #469 as well

**Comment by lukeapage** [comment]
thanks!
